library/std/src/sys_common/wtf8.rs

```diff
@@ -704,12 +762,18 @@ impl Wtf8 {
 
     #[inline]
     pub fn to_ascii_lowercase(&self) -> Wtf8Buf {
-        Wtf8Buf { bytes: self.bytes.to_ascii_lowercase() }
+        Wtf8Buf {
+            bytes: self.bytes.to_ascii_lowercase(),
+            is_known_utf8: self.next_surrogate(0).is_none(),
```

Comment:
does it make sense to scan eagerly? maybe this should just be `false` instead?

Reply:
Makes sense.